zwave_js_server/model/controller/firmware.py

```diff
@@ -1,6 +1,36 @@
 """Provide a model for Z-Wave controller firmware."""
+from dataclasses import dataclass
 from enum import IntEnum
-from typing import TypedDict
+from typing import Optional, TypedDict
+
+from ...util.helpers import convert_bytes_to_base64
+
+
+class ControllerFirmwareUpdateDataDataType(TypedDict, total=False):
+    """Represent a controller firmware update data dict type."""
+
+    filename: str  # required
```

Comment:
We could import `Required` from `typing_extensions` and use here. It's available by default in Python 3.11.

```py
from typing_extensions import Required
```

https://peps.python.org/pep-0655/

Reply:
should I do that here or do that across the board in another PR? I was reviewing the PEPs this weekend and I want to switch to union typing and generic typing in addition to more strongly typing our typeddicts. It's a lot so would have to be broken up over several PRs

Reply:
We can do it in one or more follow ups. :+1: